MalApi/MyAnimeListApi.cs

```diff
@@ -72,6 +78,40 @@ private HttpRequestMessage InitNewRequest(string uri, HttpMethod method)
             return request;
         }
 
+        private HttpRequestMessage InitNewRequestWithCredentials(string user, string password, string uri, HttpMethod method, UpdateObjectBase updateInfo = null)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(method, uri);
+
+            // Requests with credentials require them to be encoded in base64
+            string credentials;
+            if (user != null && password != null)
+            {
+                var plainTextBytes = Encoding.UTF8.GetBytes(user + ":" + password);
+                credentials =  Convert.ToBase64String(plainTextBytes);
+            }
+            else
+            {
+                throw new Exception("Invalid input credentials format (both username and password have to be input OR just the base64 representation");
```

Comment:
Instead of Exception, ArgumentNullException would be more appropriate. In order to call the ArgumentNullException constructor that takes a parameter name, you'll have to check user and password one at a time.

Reply:
Also, I am confused by the "OR just the base64 representation" in the error message.

Reply:
The error message part was a leftover from earlier ideas.